extensions/amp-story/0.1/page-advancement.js

```diff
@@ -0,0 +1,538 @@
+    this.advancementModes_ = advancementModes;
+  }
+
+  /** @override */
+  addProgressListener(progressListener) {
+    this.advancementModes_.forEach(advancementMode => {
+      advancementMode.addProgressListener(progressListener);
+    });
+  }
+
+  /** @override */
+  addAdvanceListener(advanceListener) {
+    this.advancementModes_.forEach(advancementMode => {
+      advancementMode.addAdvanceListener(advanceListener);
+    });
+  }
+
+  /** @override */
+  addPreviousListener(previousListener) {
+    this.advancementModes_.forEach(advancementMode => {
+      advancementMode.addPreviousListener(previousListener);
+    });
+  }
+
+  /** @override */
+  start() {
+    super.start();
+    this.advancementModes_.forEach(advancementMode => {
+      advancementMode.start();
+    });
+  }
+
+  /** @override */
+  stop() {
+    super.stop();
+    this.advancementModes_.forEach(advancementMode => {
+      advancementMode.stop();
+    });
+  }
+}
+
+
+/**
+ * Always provides a progress of 1.0.  Advances when the user taps the rightmost
+ * 75% of the screen; triggers the previous listener when the user taps the
+ * leftmost 25% of the screen.
+ */
+class ManualAdvancement extends AdvancementConfig {
+  /**
+   * @param {!Element} element The element that, when clicked, can cause
+   *     advancing to the next page or going back to the previous.
+   */
+  constructor(element) {
+    super();
+    this.element_ = element;
+    this.clickListener_ = this.maybePerformNavigation_.bind(this);
+  }
+
+  /** @override */
+  start() {
+    super.start();
+    this.element_.addEventListener('click', this.clickListener_, true);
+  }
+
+  /** @override */
+  stop() {
+    super.stop();
+    this.element_.removeEventListener('click', this.clickListener_, true);
+  }
+
+  /** @override */
+  getProgress() {
+    return 1.0;
+  }
+
+  /**
+   * Determines whether a click should be used for navigation.  Navigate should
+   * occur unless the click is on the system layer, or on an element that
+   * defines on="tap:..."
+   * @param {!Event} e 'click' event.
+   * @return {boolean} true, if the click should be used for navigation.
+   * @private
+   */
+  isNavigationalClick_(e) {
+    return !closest(dev().assertElement(e.target), el => {
+      return hasTapAction(el);
+    }, /* opt_stopAt */ this.element_);
+  }
+
+  /**
+   * Performs a system navigation if it is determined that the specified event
+   * was a click intended for navigation.
+   * @param {!Event} event 'click' event
+   * @private
+   */
+  maybePerformNavigation_(event) {
+    if (!this.isNavigationalClick_(event)) {
+      // If the system doesn't need to handle this click, then we can simply
+      // return and let the event propagate as it would have otherwise.
+      return;
+    }
+
+    event.stopPropagation();
+
+    // TODO(newmuis): This will need to be flipped for RTL.
+    const elRect = this.element_./*OK*/getBoundingClientRect();
```

Comment:
Marking this as `/*OK*/` since it replaces the previous calls to `offsetLeft` and `offsetWidth` that were previously marked `/*OK*/`.